### PR TITLE
Create a button to delete references

### PR DESCRIPTION
**Pull request by jwq1**
As a user
I want to delete references
so that I can remove outdated information

PR Scope: Add a delete button to each reference.

### CODE REVIEW

File: final-project/static/scripts.js [hunk 1]
```diff
@@ -528,6 +529,69 @@ function insertSaveButton() {
...
+    );
+  } else {
+    // Create a delete button.
+    var deleteButton = document.createElement('input');
+    deleteButton.setAttribute('value', 'x');
+    deleteButton.setAttribute('type', 'button');
+    deleteButton.setAttribute('class', 'delete-element');
+
+    if (appendToTheseElements.id != null) {
+      deleteButton.setAttribute('id', appendToTheseElements.id + '-delete-btn');
```
Comment (hdavidzhu) [inline final-project/static/scripts.js:573]: Seem like a large method; could you break it up?

Comment (hdavidzhu) [reply final-project/static/scripts.js:573]: https://www.youtube.com/watch?v=L2m-S0Pj_Xk

The example is a bit contrived, but imagine for a more complex project that didn't separate out these tasks -> it'll be more rigid to use, fix, and evolve issues.

File: final-project/static/scripts.js [hunk 2]
```diff
@@ -528,6 +529,69 @@ function insertSaveButton() {
...
+  });
+}
+
+// TODO Listen for clicks on the delete button.
+function listenForDelete() {
+  var deleteButtons = querySelectorAll('.delete-element');
+}
+
+// TODO Delete data from the database.
+function deleteFromDatabase(dataToDelete) {
```
Comment (hdavidzhu) [inline final-project/static/scripts.js:591]: I hope this won't be writing directly to DB. It's not necessary for the client to understand the server-side concepts of data. A `delete` would suffice.

Comment (jwq1) [reply final-project/static/scripts.js:591]: No, it will only tell the server what to delete

**Pull request merged**